packages/ui/primitives/theme-switcher.tsx

```diff
@@ -0,0 +1,33 @@
+import * as ToggleGroup from '@radix-ui/react-toggle-group';
+import { Monitor, MoonStar, Sun } from 'lucide-react';
+
+interface IProps {
+  theme?: string;
+  // eslint-disable-next-line no-unused-vars
+  setTheme: (theme: string) => void;
+}
+
+export const ThemeSwitcher = (props: IProps) => {
+  const { theme, setTheme } = props;
+
+  return (
+    <ToggleGroup.Root
```

Comment:
Feel like we can solve this with some cheeky tailwind and framer motion. Something tells me that the `layoutId` will play a part in it.

Reply:
I'm a bit new to framer motion, but understood the basic stuff on how `motion.div` and all works. Plus ChatGPT is useful sometimes xD

Just sent a commit below to use framer motion for this functionality :)